test/functional/test-url-replacements.js

```
@@ -112,6 +118,24 @@ describe('UrlReplacements', () => {
     return win;
   }
 
+  function buildClickEvent(opt_href, opt_tagName) {
+    return {
+      clientX: 123, clientY: 456,
+      target: {
+        tagName: opt_tagName ? opt_tagName : 'A',
+        getAttribute: function(name) {
+          if (name == 'href') {
+            return opt_href !== undefined ? opt_href :
+              'r=RANDOM&nx=CLICK_X&ny=CLICK_Y&plt=PAGE_LOAD_TIME&a=b';
+          } else if (name == 'data-orig-href') {
+            return undefined;
+          }
+          fail('unknown attribute: ' + name);
```

Does this do what we expect?  I'm just feeling twice shy about when Chai actually _does_ fail the test, versus executing and silently ignoring the result.  Worried that you might need to `return fail()` or something.